src/libraries/Microsoft.PowerFx.Interpreter/CustomFunction/CustomTexlFunction.cs

```diff
@@ -50,5 +54,32 @@ public override bool IsLazyEvalParam(int index)
         {
             return LamdaParamMask.TestBit(index);
         }
+
+        public override bool CheckTypes(CheckTypesContext context, TexlNode[] args, DType[] argTypes, IErrorContainer errors, out DType returnType, out Dictionary<TexlNode, DType> nodeToCoercedTypeMap)
+        {
+            Contracts.AssertValue(args);
+            Contracts.AssertAllValues(args);
+            Contracts.AssertValue(argTypes);
+            Contracts.Assert(args.Length == argTypes.Length);
+            Contracts.AssertValue(errors);
+
+            var isValid = base.CheckTypes(context, args, argTypes, errors, out returnType, out nodeToCoercedTypeMap);
+
+            // Check if all record names of args exist against return type and if its possible to coerce.
+            if (isValid && returnType.IsRecord)
```

Comment:
>returnType.IsRecord [](http://example.com/codeflow?start=27&length=19)

Why do we check ReturnType? What if function takes in a record but doesn't return record?

Reply:
Good point. remove this check. Add test for non record.